backtrace-library/src/androidTest/java/backtraceio/library/BacktraceClientMetricsTest.java

```diff
@@ -0,0 +1,938 @@
+     * For best results run under low CPU load and low memory utilization conditions.
+     */
+
+    @Before
+    public void setUp() {
+        context = InstrumentationRegistry.getContext();
+        credentials = new BacktraceCredentials("https://example-endpoint.com/", "");
+        BacktraceDatabase database = new BacktraceDatabase(context, context.getFilesDir().getAbsolutePath());
+
+        backtraceClient = new BacktraceClient(context, credentials, database);
+
+        BacktraceLogger.setLevel(LogLevel.DEBUG);
+    }
+
+    public class MockRequestHandler implements EventsRequestHandler {
+        public int numAttempts = 0;
+        public int statusCode = 200;
+        public String lastEventPayloadJson;
+
+        @Override
+        public EventsResult onRequest(EventsPayload data) {
+            String eventsPayloadJson = BacktraceSerializeHelper.toJson(data);
+            lastEventPayloadJson = eventsPayloadJson;
+            numAttempts++;
+
+            BacktraceResultStatus status;
+            if (statusCode == 200) {
+                status = BacktraceResultStatus.Ok;
+            } else {
+                status = BacktraceResultStatus.ServerError;
+            }
+            return new EventsResult(data, eventsPayloadJson, status, statusCode);
+        }
+    }
+
+    @Test
+    public void disableAutoSend() {
+        backtraceClient.metrics.enable(new BacktraceMetricsSettings(universeName, token, defaultBaseUrl, 0));
+        MockRequestHandler mockRequestHandler = new MockRequestHandler();
+        backtraceClient.metrics.setUniqueEventsRequestHandler(mockRequestHandler);
+
+        backtraceClient.metrics.addUniqueEvent(uniqueAttributeName[0]);
+        try {
+            sleep(100);
+        } catch (Exception e) {
+            fail(e.toString());
+        }
+        assertEquals(0, mockRequestHandler.numAttempts);
+    }
+
+    @Test
+    public void uploadSummedEventsManual() {
+        final Waiter waiter = new Waiter();
+
+        backtraceClient.metrics.enable(new BacktraceMetricsSettings(universeName, token, defaultBaseUrl, 0));
+        MockRequestHandler mockRequestHandler = new MockRequestHandler();
+        backtraceClient.metrics.setSummedEventsRequestHandler(mockRequestHandler);
+
+        backtraceClient.setSummedEventsOnServerResponse(new EventsOnServerResponseEventListener() {
+            @Override
+            public void onEvent(EventsResult result) {
+                assertEquals(0, result.getEventsPayload().getDroppedEvents());
+                String eventsPayloadJson = BacktraceSerializeHelper.toJson(result.getEventsPayload());
+                assertNotEquals(0, eventsPayloadJson.length());
+                assertEquals(BacktraceResultStatus.Ok, result.status);
+                waiter.resume();
+            }
+        });
+
+        backtraceClient.metrics.addSummedEvent(summedEventName);
+
+        assertEquals(1, backtraceClient.metrics.getSummedEvents().size());
```

Comment:
I would recommend to do assertions in the end of the test.

Reply:
We are asserting that we should have one summed event added to the queue before calling `send` method, and that after `send` method summed event queue should be empty. Should we still remove this assertion @konraddysput?

Reply:
It's also a good practice to do assertion on the end of the unit test: Arrange, Act and Assert (AAA) Pattern. I would recommend to not test everything if we already have a unit test for it.

Reply:
Done, thank you!